sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClientTest.java

```diff
@@ -370,17 +488,48 @@ void scheduleMessage() {
         when(connection.createSendLink(eq(ENTITY_NAME), eq(ENTITY_NAME), any(AmqpRetryOptions.class)))
             .thenReturn(Mono.just(sendLink));
         when(sendLink.getLinkSize()).thenReturn(Mono.just(MAX_MESSAGE_LENGTH_BYTES));
-        when(managementNode.schedule(eq(message), eq(instant), any(Integer.class), any()))
+        when(managementNode.schedule(eq(message), eq(instant), any(Integer.class), any(), isNull()))
             .thenReturn(just(sequenceNumberReturned));
 
         // Act & Assert
         StepVerifier.create(sender.scheduleMessage(message, instant))
             .expectNext(sequenceNumberReturned)
             .verifyComplete();
 
-        verify(managementNode).schedule(message, instant, MAX_MESSAGE_LENGTH_BYTES, LINK_NAME);
+        verify(managementNode).schedule(message, instant, MAX_MESSAGE_LENGTH_BYTES, LINK_NAME, null);
+    }
+
+    @Test
+    void scheduleMessageWithTransaction() {
+        // Arrange
+        long sequenceNumberReturned = 10;
+        Instant instant = mock(Instant.class);
+        ArgumentCaptor<Instant> instantCapture = ArgumentCaptor.forClass(Instant.class);
+        ArgumentCaptor<Integer> lengthCapture = ArgumentCaptor.forClass(Integer.class);
+        ArgumentCaptor<String> linkNameCapture = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<ServiceBusTransactionContext> transactionCapture = ArgumentCaptor.forClass(ServiceBusTransactionContext.class);
+
+        when(connection.createSendLink(eq(ENTITY_NAME), eq(ENTITY_NAME), any(AmqpRetryOptions.class)))
+            .thenReturn(Mono.just(sendLink));
+        when(sendLink.getLinkSize()).thenReturn(Mono.just(MAX_MESSAGE_LENGTH_BYTES));
+        when(managementNode.schedule(eq(message), eq(instant), any(Integer.class), any(), any(ServiceBusTransactionContext.class)))
+            .thenReturn(just(sequenceNumberReturned));
+
+        // Act & Assert
+        StepVerifier.create(sender.scheduleMessage(message, instant, transactionContext))
+            .expectNext(sequenceNumberReturned)
+            .verifyComplete();
+
+        verify(managementNode).schedule(singleSBMessageCaptor.capture(), instantCapture.capture(), lengthCapture.capture(), linkNameCapture.capture(), transactionCapture.capture());
+
+        Assertions.assertEquals(message, singleSBMessageCaptor.getValue());
```

Comment:
you don't need to capture them. if the arguments didn't match, it would return null and fail teh tests.